pkg/policy/hash_bench_test.go

```diff
@@ -0,0 +1,95 @@
+package policy
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/bluele/gcache"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/aquasecurity/trivy-operator/pkg/plugins/trivy"
+	"github.com/aquasecurity/trivy-operator/pkg/utils"
+	"github.com/aquasecurity/trivy/pkg/fanal/types"
+)
+
+var (
+	k8sresources = []string{
+		"Pod",
+		"Deployment",
+		"DaemonSet",
+		"StatefulSet",
+		"ReplicaSet",
+	}
+)
+
+type testConfig struct {
+	builtInPolicies  bool
+	embeddedPolicies bool
+}
+
+func newTestConfig(builtInPolicies bool) testConfig {
+	return testConfig{builtInPolicies: builtInPolicies}
+}
+
+// GetUseBuiltinRegoPolicies return trivy config which associated to configauditreport plugin
+func (tc testConfig) GetUseBuiltinRegoPolicies() bool {
+	return tc.builtInPolicies
+}
+
+// GetUseBuiltinRegoPolicies return trivy config which associated to configauditreport plugin
+func (tc testConfig) GetUseEmbeddedRegoPolicies() bool {
+	return tc.embeddedPolicies
+}
+
+// GetSupportedConfigAuditKinds list of supported kinds to be scanned by the config audit scanner
+func (tc testConfig) GetSupportedConfigAuditKinds() []string {
+	return utils.MapKinds(strings.Split(trivy.SupportedConfigAuditKinds, ","))
+}
+
+func (tc testConfig) GetSeverity() string {
+	return trivy.KeyTrivySeverity
+}
+
+func benchmarkHash10000(b *testing.B, p *Policies) {
+	b.ResetTimer()
+	b.ReportAllocs()
+
+	for i := 0; i < 10000; i++ {
```

Comment:
This has to be set to `b.N`
```suggestion
	for i := 0; i < b.N; i++ {
```


You can control the iterations and time with `-benchtime` and `-count` flags.

Reply:
I'm also running into a strange error when I run the benchmarks

```shell
goos: darwin
goarch: arm64
pkg: github.com/aquasecurity/trivy-operator/pkg/policy
cpu: Apple M1 Pro
BenchmarkPolicies_Hash
[controller-runtime] log.SetLogger(...) was never called; logs will not be displayed.
Detected at:
        >  goroutine 851 [running]:
        >  runtime/debug.Stack()
        >       /Users/simarpreetsingh/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.1.darwin-arm64/src/runtime/debug/stack.go:26 +0x64
        >  sigs.k8s.io/controller-runtime/pkg/log.eventuallyFulfillRoot()
        >       /Users/simarpreetsingh/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.20.4/pkg/log/log.go:60 +0xf4
        >  sigs.k8s.io/controller-runtime/pkg/log.(*delegatingLogSink).WithName(0x14000d5e080, {0x10793d1a9, 0x1d})
        >       /Users/simarpreetsingh/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.20.4/pkg/log/deleg.go:147 +0x34
        >  github.com/go-logr/logr.Logger.WithName(...)
        >       /Users/simarpreetsingh/go/pkg/mod/github.com/go-logr/logr@v1.4.2/logr.go:345
        >  github.com/aquasecurity/trivy-operator/pkg/policy.(*policyLoader).GetPoliciesAndBundlePath(0x14000eba000)
        >       /Users/simarpreetsingh/repos/trivy-operator/pkg/policy/loader.go:54 +0x54
        >  github.com/aquasecurity/trivy-operator/pkg/policy.(*Policies).loadPolicies(0x140002defd0, {0x1078ad1f3?, 0x140028ce9b0?})
        >       /Users/simarpreetsingh/repos/trivy-operator/pkg/policy/policy.go:227 +0x1b0
        >  github.com/aquasecurity/trivy-operator/pkg/policy.(*Policies).Hash(0x140002defd0, {0x1078ad1f3, 0xa})
        >       /Users/simarpreetsingh/repos/trivy-operator/pkg/policy/policy.go:168 +0x44
        >  github.com/aquasecurity/trivy-operator/pkg/policy.benchmarkHash10000(0x1400282e288, 0x140002defd0)
        >       /Users/simarpreetsingh/repos/trivy-operator/pkg/policy/hash_bench_test.go:60 +0x8c
        >  github.com/aquasecurity/trivy-operator/pkg/policy.BenchmarkPolicies_Hash(0x1400282e288)
        >       /Users/simarpreetsingh/repos/trivy-operator/pkg/policy/hash_bench_test.go:72 +0x220
        >  testing.(*B).runN(0x1400282e288, 0x209d98)
        >       /Users/simarpreetsingh/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.1.darwin-arm64/src/testing/benchmark.go:202 +0x198
        >  testing.(*B).launch(0x1400282e288)
        >       /Users/simarpreetsingh/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.1.darwin-arm64/src/testing/benchmark.go:333 +0x178
        >  created by testing.(*B).doBench in goroutine 1
        >       /Users/simarpreetsingh/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.1.darwin-arm64/src/testing/benchmark.go:275 +0x68
BenchmarkPolicies_Hash-10                       17399216               658.7 ns/op            81 B/op          5 allocs/op
<snip>
```

Reply:
But overall after making this correction the results seem in line to me:

`-benchtime=10s -count=3 `

```shell
BenchmarkPolicies_Hash-10                       15214363               672.7 ns/op            82 B/op          5 allocs/op
BenchmarkPolicies_Hash-10                       14770934               679.0 ns/op            82 B/op          5 allocs/op
BenchmarkPolicies_Hash-10                       14872353               681.1 ns/op            82 B/op          5 allocs/op
BenchmarkPolicies_HashWithoutCache-10               2728           3736391 ns/op         8060024 B/op        982 allocs/op
BenchmarkPolicies_HashWithoutCache-10               3325           3696082 ns/op         8058525 B/op        980 allocs/op
BenchmarkPolicies_HashWithoutCache-10               2926           3543557 ns/op         8059487 B/op        981 allocs/op

```

Reply:
> I'm also running into a strange error when I run the benchmarks

fixed https://github.com/aquasecurity/trivy-operator/pull/2526/commits/d2ad385643c6750a38fd04c13011752462e650e4

Reply:
> This has to be set to `b.N`
> 
> You can control the iterations and time with `-benchtime` and `-count` flags.

fixed https://github.com/aquasecurity/trivy-operator/pull/2526/commits/d2ad385643c6750a38fd04c13011752462e650e4